elide-async/src/main/java/com/yahoo/elide/async/export/TableExporter.java

```diff
@@ -0,0 +1,98 @@
+import com.yahoo.elide.async.models.QueryType;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.core.exceptions.InvalidValueException;
+import com.yahoo.elide.graphql.GraphQLRequestScope;
+import com.yahoo.elide.request.EntityProjection;
+import com.yahoo.elide.security.User;
+
+import io.reactivex.Observable;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.UUID;
+
+/**
+ * Class for Table Export functionality.
+ */
+public class TableExporter {
+
+    private Elide elide;
+    private String apiVersion;
+    private User user;
+    private GraphQLParser graphQLParser;
+
+    public TableExporter(Elide elide, String apiVersion, User user) {
+        this(elide, apiVersion, user, new GraphQLParser(elide, apiVersion));
+    }
+
+    public TableExporter(Elide elide, String apiVersion, User user, GraphQLParser graphQLParser) {
+        this.elide = elide;
+        this.apiVersion = apiVersion;
+        this.user = user;
+        this.graphQLParser = graphQLParser;
+    }
+
+    /**
+     * Exports the Data based on AsyncQuery.
+     * @param query AsyncQuery object.
+     * @return Observable of PersistentResource.
+     */
+    public Observable<PersistentResource> export(AsyncQuery query) {
+        Observable<PersistentResource> results = Observable.empty();
+
+        UUID requestId = UUID.fromString(query.getRequestId());
+
+        try (DataStoreTransaction tx = elide.getDataStore().beginTransaction()) {
+            elide.getTransactionRegistry().addRunningTransaction(requestId, tx);
+
+            EntityProjection projection = null;
+            RequestScope requestScope = null;
+
+            if (query.getQueryType().equals(QueryType.GRAPHQL_V1_0)) {
+                projection = graphQLParser.parse(query);
+                //TODO - we need to add the baseUrlEndpoint to the queryObject.
+                //TODO - Can we have projectionInfo as null?
+                requestScope = new GraphQLRequestScope("", tx, user, apiVersion, elide.getElideSettings(), null,
+                        requestId);
+            } else {
+                //TODO - Add JSON Support
+                throw new InvalidValueException("QueryType not supported");
+            }
+
+            if (projection != null) {
+                results = PersistentResource.loadRecords(projection, Collections.emptyList(), requestScope);
+            }
+
+            tx.preCommit();
+            requestScope.runQueuedPreSecurityTriggers();
+            requestScope.getPermissionExecutor().executeCommitChecks();
+
+            tx.flush(requestScope);
+
+            requestScope.runQueuedPreCommitTriggers();
+
+            elide.getAuditLogger().commit();
+            tx.commit(requestScope);
+
+            requestScope.runQueuedPostCommitTriggers();
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
```

Comment:
IOExceptions are handled by throwing a TransactionException in Elide.  We should be consistent here.